flyway-community-db-support/src/main/java/org/flywaydb/community/database/clickhouse/ClickHouseDatabase.java

```diff
@@ -0,0 +1,112 @@
+package org.flywaydb.community.database.clickhouse;
+
+import org.flywaydb.core.api.configuration.Configuration;
+import org.flywaydb.core.internal.database.base.Database;
+import org.flywaydb.core.internal.database.base.Table;
+import org.flywaydb.core.internal.jdbc.JdbcConnectionFactory;
+import org.flywaydb.core.internal.jdbc.StatementInterceptor;
+import org.flywaydb.core.internal.util.StringUtils;
+
+import java.sql.Connection;
+
+public class ClickHouseDatabase extends Database<ClickHouseConnection>
+{
+    @Override
+    public boolean useSingleConnection() {
+        return true;
+    }
+
+    public ClickHouseDatabase(Configuration configuration, JdbcConnectionFactory jdbcConnectionFactory, StatementInterceptor statementInterceptor) {
+        super(configuration, jdbcConnectionFactory, statementInterceptor);
+    }
+
+    @Override
+    protected ClickHouseConnection doGetConnection(Connection connection) {
+        return new ClickHouseConnection(this, connection);
+    }
+
+    @Override
+    public void ensureSupported() {
+    }
+
+    @Override
+    public boolean supportsDdlTransactions() {
+        return false;
+    }
+
+    @Override
+    public boolean supportsMultiStatementTransactions() {
+        return false;
+    }
+
+    @Override
+    public String getBooleanTrue() {
+        return "1";
+    }
+
+    @Override
+    public String getBooleanFalse() {
+        return "0";
+    }
+
+    @Override
+    public boolean catalogIsSchema() {
+        return true;
+    }
+
+    @Override
+    public String getRawCreateScript(Table table, boolean baseline) {
+        String clusterName = configuration.getPluginRegister().getPlugin(ClickHouseConfigurationExtension.class).getClusterName();
+
+        if (StringUtils.hasText(clusterName)) {
+            return "CREATE TABLE IF NOT EXISTS " + table + "_local ON CLUSTER " + clusterName + "(" +
```

Comment:
This concatination is wrong. Result string:
```
CREATE TABLE IF NOT EXISTS "metrics_db"."flyway_schema_history"_local ON CLUSTER test_shard_four_shards_localhost(    installed_rank Int32,    version String,    description String,    type String,    script String,    checksum Nullable(Int32),    installed_by String,    installed_on DateTime DEFAULT now(),    execution_time Int32,    success Bool) ENGINE = ReplicatedMergeTree(   '/clickhouse/tables/{shard}/flyway_schema_history',   '{replica}' ) PARTITION BY tuple() ORDER BY (installed_rank)
```
Server error:
```
Code: 62. DB::Exception: Syntax error: failed at position 64 ('_local'): _local ON CLUSTER test_shard_four_shards_localhost(    installed_rank Int32,    version String,    description String,    type String,    script String,    chec. Expected one of: token, Dot, UUID, ON, OpeningRoundBracket, storage definition, ENGINE, PARTITION BY, PRIMARY KEY, ORDER BY, SAMPLE BY, TTL, EMPTY AS, AS, COMMENT, INTO OUTFILE, FORMAT, SETTINGS, end of query. (SYNTAX_ERROR) (version 22.9.6.20 (official build))
, server ClickHouseNode [uri=http://localhost:49635/metrics_db]@206826844
```

Reply:
This error is repeatable on my branch (for now not access on github). I was port you PR to [flyway-8.5.13](https://github.com/flyway/flyway/releases/tag/flyway-8.5.13) because [spring-boot 2.7.x](https://github.com/spring-projects/spring-boot/blob/2.7.x/spring-boot-project/spring-boot-dependencies/build.gradle#L347) depend on this version of library.

Main difference 8.5.13 & 9.15.1 is how we access clusterName. In 8.5.13
```java
String clusterName = PluginRegister.getPlugin(ClickHouseConfigurationExtension.class).getClusterName();
```
Anyway if you have this bug in [flyway-9.15.1](https://github.com/flyway/flyway/releases/tag/flyway-9.15.1) this should help you
```java
String tableName = table.getName();
String localTableNameQuoted = quote(tableName + "_local");
```

Reply:
I've reworked cluster table creation and will push it today. I recommend wait for the fix, because there will be no Distributed table because it doesn't support ClickHouse mutations and flyway "repair" command doesn't work.

Reply:
> This error is repeatable on my branch (for now not access on github). I was port you PR to [flyway-8.5.13](https://github.com/flyway/flyway/releases/tag/flyway-8.5.13) because [spring-boot 2.7.x](https://github.com/spring-projects/spring-boot/blob/2.7.x/spring-boot-project/spring-boot-dependencies/build.gradle#L347) depend on this version of library.

I would suggest using "flyway.version" variable in your Spring Boot app to use actual flyway build and avoid backporting.

Reply:
Thank you. That works